src/client.ts

```diff
@@ -180,6 +192,73 @@ export default class Client {
     );
   }
 
+  private async setupCustomGemfile() {
+    // If we're working on the ruby-lsp itself, we can't create a custom Gemfile or we'd be trying to activate the same
+    // gem twice
+    if (this.workingFolder.endsWith("ruby-lsp")) {
+      return;
+    }
+
+    // Create the .ruby-lsp directory if it doesn't exist
+    const rubyLspDirectory = path.join(this.workingFolder, ".ruby-lsp");
+
+    if (!fs.existsSync(rubyLspDirectory)) {
+      fs.mkdirSync(rubyLspDirectory);
+    }
+
+    // Ignore the .ruby-lsp directory automatically
+    const gitIgnorePath = path.join(this.workingFolder, ".git/info/exclude");
+
+    if (!(await readFile(gitIgnorePath)).includes(".ruby-lsp")) {
+      fs.appendFileSync(gitIgnorePath, ".ruby-lsp");
+    }
```

Comment:
The folder seemed to exist - at least for projects I tested on. But given that there are so many checks we need to do (is project using git, does `.git/info` exist, does `.git/info/exclude` exist), wouldn't it be easier and less error prone to just create the `.ruby-lsp/.gitignore` file with `*`?

Reply:
Sure, if that's easier, you can go with that. However my suggestion from earlier, expanded a little bit, should cover most of this in a few commands:
```ruby
const gitInfoFolder = path.join(rubyLspDirectory, ".git", "info");
const excludeFile = path.join(gitInfoFolder, "exclude");

fs.mkdirSync(gitInfoFolder, { recursive: true }); // create folders as needed
fs.appendFileSync(excludeFile, "", "a"); // touch the exclude file to ensure it exists
if (!fs.readFileSync(excludeFile).includes(".ruby-lsp") {
  fs.appendFileSync(excludeFile, ".ruby-lsp");
}
```
This would end up creating a `.git/info` folder even if the project isn't using `.git`, but I guess that's not the end of the world.

Reply:
I think it's simpler to use the `.gitignore` file with `*`. It does the job, keeps everything pertinent to the Ruby LSP in the same folder and is less surprising for projects not using git. I changed it in the latest commit.